### PR TITLE
Add support for 2 other OAuth providers

### PR DESCRIPTION
**Pull request by julsemaan**
Adds support for the following authentication providers : 
- Windows Live
- LinkedIn

Also rewrites minor parts of the oauth2 in the portal to be more flexible

### CODE REVIEW

File: NEWS.asciidoc [hunk 1]
```diff
@@ -18,6 +18,7 @@ New Features
 ++++++++++++
 
 * Added support for Ruckus External Captive Portal
+* Added support for LinkedIn and Windows Live as authentication source.
```
Comment (extrafu) [inline NEWS.asciidoc]: as authentication sources

File: docs/PacketFence_Administration_Guide.asciidoc [hunk 2]
```diff
@@ -2171,6 +2173,33 @@ Once you have your information, Once you have your information, you need to conf
 
 Moreover, don't forget to add GitHub as a registration mode from your portal profile definition, available from *Configuration -> Portal Profiles and Pages*.
 
+LinkedIn
+^^^^^^^^
+To use LinkedIn, you also need an API code and a secret key.  To get one, you need to create an App here: https://developer.linkedin.com/.  When you create your App, make sure you input the following as the Callback URL
```
Comment (extrafu) [inline docs/PacketFence_Administration_Guide.asciidoc]: you specify the following

File: docs/PacketFence_Administration_Guide.asciidoc [hunk 3]
```diff
@@ -2171,6 +2173,33 @@ Once you have your information, Once you have your information, you need to conf
...
+https://YOUR_PORTAL_HOSTNAME/oauth2/linkedin
+
+Of course, replace the hostname with the values from `general.hostname` and `general.domain`.
+
+Once you have your information, Once you have your information, you need to configure the OAuth2 provider. This can be done by adding a LinkedIn OAuth2 authentication source from *Configuration -> Sources*.
```
Comment (extrafu) [inline docs/PacketFence_Administration_Guide.asciidoc]: +Once you have your information, Once you have your information => Once you have your information

File: docs/PacketFence_Administration_Guide.asciidoc [hunk 4]
```diff
@@ -2171,6 +2173,33 @@ Once you have your information, Once you have your information, you need to conf
...
+
+Moreover, don't forget to add LinkedIn as a registration mode from your portal profile definition, available from *Configuration -> Portal Profiles and Pages*.
+
+Also, LinkedIn requires a 'state' parameter for the authorization URL. If you modify it, make sure to add it at the end of your URL.
+
+Windows Live
+^^^^^^^^^^^^
+To use Windows live, you also need an API code and a secret key.  To get one, you need to create an App here: https://account.live.com/developers/applications.  When you create your App, make sure you input the following as the Callback URL
```
Comment (extrafu) [inline docs/PacketFence_Administration_Guide.asciidoc]: you specify

File: docs/PacketFence_Administration_Guide.asciidoc [hunk 5]
```diff
@@ -2171,6 +2173,33 @@ Once you have your information, Once you have your information, you need to conf
...
+https://YOUR_PORTAL_HOSTNAME/oauth2/live
+
+Of course, replace the hostname with the values from `general.hostname` and `general.domain`.
+
+Once you have your information, Once you have your information, you need to configure the OAuth2 provider. This can be done by adding a Live OAuth2 authentication source from *Configuration -> Sources*.
```
Comment (extrafu) [inline docs/PacketFence_Administration_Guide.asciidoc]: Drinking problem? ;-)

Comment (julsemaan) [inline docs/PacketFence_Administration_Guide.asciidoc]: No. Just copy paste master :)

File: html/captive-portal/lib/captiveportal/PacketFence/Controller/Oauth2.pm [hunk 6]
```diff
@@ -156,26 +165,48 @@ sub oauth2Result : Path : Args(1) {
...
+                    );
+                    $pid = $json_text->{email};
+                } elsif ($provider eq 'facebook') {
+                    $logger->info(
+                        "OAuth2 successfull, register and release for username $json_text->{username}"
+                    );
+                    $pid = $json_text->{username} . '@facebook.com';
+                } elsif ($provider eq 'live'){
+                    $logger->info(
+                        "OAuth2 successfull, register and release for username $json_text->{emails}->{account}"
```
Comment (extrafu) [inline html/captive-portal/lib/captiveportal/PacketFence/Controller/Oauth2.pm]: Wouldn't it be nice to put something like:

$logger->info("OAuth2 ($provider) succesfull, ...") 

for all of the above cases?

File: lib/pf/Authentication/Source/LiveSource.pm [hunk 7]
```diff
@@ -0,0 +1,84 @@
+package pf::Authentication::Source::LiveSource;
```
Comment (extrafu) [inline lib/pf/Authentication/Source/LiveSource.pm]: IMO, we should call it "WindowsLiveSource", but that can be just me :)

Comment (julsemaan) [inline lib/pf/Authentication/Source/LiveSource.pm]: I taught of it but their service is 'live' but I agree it can be confusing.

Comment (extrafu) [inline lib/pf/Authentication/Source/LiveSource.pm]: One "argument" against "Live" would be Windows Live or LiveJournal?

Comment (julsemaan) [inline lib/pf/Authentication/Source/LiveSource.pm]: True. I'll change it monday

**Comment by fdurand** [comment]
It looks ok for me, i propose to merge

**Comment by julsemaan** [comment]
Had to merge manually
Commit c48bc9f14d7a53dcd77673b73ec9b2692c93ceab
Closing PR

**Pull request closed without merging**